uoyweek.cpp

```diff
@@ -30,47 +46,49 @@ class Term {
 
     string toString() {
         char yearStr[128];
-        //Assign result of asctime to string due to  weird memory management of asctime
+        /* Assign result of asctime to string due to  weird memory management of
+         * asctime */
         char start_string[128];
         strncpy(start_string, asctime(&start), 128);
         char end_string[128];
         strncpy(end_string, asctime(&end), 128);
-        //sprintf is fine for actual values but results of asctime, etc. should be put in string in memory first
-        sprintf(yearStr, "%s (%d-%d) %s %s", code.c_str(), ((&start)->tm_year+1900), ((&end)->tm_year+1900), start_string, end_string);
+        /* sprintf is fine for actual values but results of asctime, etc. should
+         * be put in string in memory first */
+        sprintf(yearStr, "%s (%d-%d) %s %s",
+                code.c_str(),
+                (&start)->tm_year + 1900,
+                (&end)->tm_year + 1900,
+                start_string,
+                end_string);
         return yearStr;
     }
 };
 
-Term * aut  = new Term("aut", (time_t) 1506297600l, (time_t) 1512086400l);
-Term * spr  = new Term("spr", (time_t) 1515369600l, (time_t) 1521158400l);
-string days[] = {
-    "sunday",
-    "monday",
-    "tuesday",
-    "wednesday",
-    "thursday",
-    "friday",
-    "saturday"
-};
+const int AUT_START = 1506297600l;
+const int AUT_END = 1512086400l;
+const int SPR_START = 1515369600l;
+const int SPR_END   = 1521158400l;
+
+Term *aut = new Term("aut", (time_t)AUT_START, (time_t)AUT_END);
+Term *spr = new Term("spr", (time_t)SPR_START, (time_t)SPR_END);
 
 time_t normaliseToMonday(time_t toNormalise) {
     struct tm *tN = gmtime(&toNormalise);
     time_t result = toNormalise;
     result -= tN->tm_sec;
-    result -= tN->tm_min * 60;
-    result -= tN->tm_hour * 3600;
-    result -= (tN->tm_wday - 1) * 24 * 3600;
+    result -= tN->tm_min * SECS_PER_MIN;
+    result -= tN->tm_hour * SECS_PER_HOUR;
+    result -= (tN->tm_wday - 1) * SECS_PER_DAY;
     return result;
 }
 
-int getWeek(time_t now, Term * currentTerm) {
+int getWeek(time_t now, Term *currentTerm) {
     time_t startNormalised = normaliseToMonday(currentTerm->startTS);
     time_t interval = now - startNormalised;
-    int weeks = interval/604800;
-    return weeks + 1;
+    return interval/SECS_PER_WEEK + 1;
```

Comment:
spacing

Reply:
`interval / SECS_PER_WEEK`?

Reply:
`a/b + c` makes reading it easier a glance, IMO.

Reply:
Aye, use brackets if you're not comfortable "relying" on operator precedence